src/Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis/FunctionalRedisCache.cs

```diff
@@ -0,0 +1,293 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis.JsonConverters;
+using Newtonsoft.Json;
+using ServiceStack.Redis;
+
+namespace Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis
+{
+	// ReSharper disable once InconsistentNaming
+	/// <summary>
+	/// Used to cache data remotely in a Redis cache.
+	/// Note that objects stored in the cache must be concrete types in order for serialization to work correctly.
+	/// </summary>
+	public class FunctionalRedisCache : IFunctionalCache
+	{
+		private readonly ConcurrentDictionary<string, SemaphoreSlim> _semaphoreSlimLookup = new ConcurrentDictionary<string, SemaphoreSlim>();
+		private static readonly JsonSerializerSettings _serializerSettings = new JsonSerializerSettings()
+		{
+			Converters = new List<JsonConverter>
```

Comment:
I could see adding the ability to inject custom converters being a feature request one day...

Reply:
I'll log an issue so I remember to come back to it